packages/babel-plugin/src/utils/css-builders.tsx

```diff
@@ -400,6 +400,58 @@ const extractLogicalExpression = (node: t.ArrowFunctionExpression, meta: Metadat
   return { css: mergeSubsequentUnconditionalCssItems(css), variables };
 };
 
+/**
+ * Manipulates the AST to ensure that CSS variables are generated correctly for negative values
+ *
+ * consider we have the following:
+ *
+ * const gridSize = 8;
+ *
+ * const LayoutRight = styled.aside`
+ *   margin-right: -${gridSize * 5}px; // A
+ *   margin-left: ${gridSize * 5}px; // B
+ *   top: -${gridSize * 5}px; // A
+ *   left: -${gridSize * 8}px; // C
+ *   right: ${gridSize * 8}px;  // D
+ *   color: red;
+ * `;
+ *
+ * In order to calculate the correct CSS custom properties, items labeled A & C need to be converted to UnaryExpressions
+ * This essentially changes these to ${-gridSize * 5}px; & ${-gridSize * 8}px respectively - resulting in correct CCS
+ * generation for these negative values.
+ *
+ * With out this manipulation you would end up with:
+ * -40px for A & B
+ * -64px for C & D
+ *
+ * With this manipulation you end up with
+ * -40px for A
+ * 40px for B
+ * -64px for C
+ * 64px for D
+ *
+ * @param nodeExpression Node we're interested in manipulating
+ * @param quasi quasi information about the current node
+ */
+const convertNegativeCssValuesToUnaryExpression = (nodeExpression: t.Expression): void => {
```

Comment:
Can we use a more specific type, eg `BinaryExpression`, or the node could be anything?

Reply:
The expression can be anything but we can do a type check on the expression on line 437 instead of `nodeExpression?.left`